AssertionError in the test_quantity_ufuncs

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->


<!-- Provide a general description of the bug. -->
Following instructions in [Try the development version](http://docs.astropy.org/en/stable/development/workflow/get_devel_version.html), I've got one failure after `python setup.py test` at the shell prompt:
```
self = <astropy.units.tests.test_quantity_ufuncs.TestUfuncHelpers object at >

    def test_coverage(self):
        """Test that we cover all ufunc's"""    
        all_np_ufuncs = set([ufunc for ufunc in np.core.umath.__dict__.values()
                             if isinstance(ufunc, np.ufunc)])
    
        all_q_ufuncs = (qh.UNSUPPORTED_UFUNCS |
                        set(qh.UFUNC_HELPERS.keys()))
        # Check that every numpy ufunc is covered.
    --> assert all_np_ufuncs - all_q_ufuncs == set()
E       AssertionError: assert {<ufunc 'erf'>} == set()
E         Extra items in the left set:
E         <ufunc 'erf'>
E         Use -v to get the full diff

astropy/units/tests/test_quantity_ufuncs.py:69: AssertionError
```
![Screenshot_20191021_101816](https://user-images.githubusercontent.com/55282885/67188300-e3280300-f3f8-11e9-9bc7-4ad724b33f06.png)

<!-- What did you expect to happen. -->


<!-- What actually happened. -->
<!-- Was the output confusing or poorly described? -->


<!-- Ideally a code example could be provided so we can run it ourselves. -->
<!-- If you are pasting code, use tripe backticks (```) around
your code snippet. -->
<!-- If necessary, sanitize your screen output to be pasted so you do not
reveal secrets like tokens and passwords. -->

<!-- Even if you do not think this is necessary, it is useful information for the maintainers.
Please run the following snippet and paste the output below:
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("Numpy", numpy.__version__)
import scipy; print("Scipy", scipy.__version__)
import astropy; print("astropy", astropy.__version__)
-->

Linux-4.19.0-6-amd64-x86_64-with-debian-10.1
Python 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]
Numpy 1.16.4
Scipy 1.3.0
astropy 4.0.dev26155

EDIT: Formatting.

--- a/astropy/units/quantity_helper/helpers.py
+++ b/astropy/units/quantity_helper/helpers.py
@@ -355,6 +355,12 @@ for ufunc in invariant_ufuncs:
 # ufuncs that require dimensionless input and and give dimensionless output
 dimensionless_to_dimensionless_ufuncs = (np.exp, np.expm1, np.exp2, np.log,
                                          np.log10, np.log2, np.log1p)
+# Default numpy does not ship an "erf" ufunc, but some versions hacked by
+# intel do.  This is bad, since it means code written for that numpy will
+# not run on non-hacked numpy.  But still, we might as well support it.
+if isinstance(getattr(np.core.umath, 'erf', None), np.ufunc):
+    dimensionless_to_dimensionless_ufuncs += (np.core.umath.erf,)
+
 for ufunc in dimensionless_to_dimensionless_ufuncs:
     UFUNC_HELPERS[ufunc] = helper_dimensionless_to_dimensionless
 

